### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by fominmaksim**
https://trello.com/c/YiknL5oU - completed the task

### CODE REVIEW

**Review by AndrewKaenko — APPROVED** [review]
Everything is checked. No questions

File: src/containers/SearchPanel.jsx [hunk 1]
```diff
@@ -1,8 +1,139 @@
+import { useState } from "react"
+import { Link } from "react-router-dom"
+import styled from "styled-components"
+
+const StyledP = styled.p`
+  color: gray;
+  text-decoration: none;
```
Comment (alexandrtovmach) [inline src/containers/SearchPanel.jsx:7]: useless

File: src/containers/SearchPanel.jsx [hunk 2]
```diff
@@ -1,8 +1,139 @@
...
+      <StyledLink
+        to={`/search-result?departureCity=${departureCityInput}&arrivalCity=${arrivalCityInput}&departureStartDate=${departureStartDateInput}&departureEndDate=${departureEndDateInput}`}
+      >
+        <ButtonSection>
+          <NextBtn>
+            <Arrow>&#8594;</Arrow>
+          </NextBtn>
+          <StyledP>Next</StyledP>
+        </ButtonSection>
+      </StyledLink>
```
Comment (alexandrtovmach) [inline src/containers/SearchPanel.jsx:135]:
```
<StyledLink
        to={`/search-result?departureCity=${departureCityInput}&arrivalCity=${arrivalCityInput}&departureStartDate=${departureStartDateInput}&departureEndDate=${departureEndDateInput}`}
      >
          <NextBtn>
            <Arrow>&#8594;</Arrow>
          </NextBtn>
          <StyledP>Next</StyledP>
      </StyledLink>
```

`<ButtonSection>` - useless. Move all styles from it to parent Styled link and remove it from layout

File: src/containers/SearchResult.jsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+          <StopsDiv>
+            <Dot />
+            <Dot />
+            <Dot />
+            <StopsP>0 Stops</StopsP>
+            <Dot />
+            <Dot />
+            <Dot />
+          </StopsDiv>
+          <FlightDurationP>{flightDuration().toString()}</FlightDurationP>
```
Comment (alexandrtovmach) [inline src/containers/SearchResult.jsx:187]:
```suggestion
          <FlightDurationP>{flightDuration()}</FlightDurationP>
```

File: src/pages/Search.jsx [hunk 4]
```diff
@@ -1,10 +1,46 @@
+// import { BrowserRouter as Router, Switch, Route } from "react-router-dom"
```
Comment (alexandrtovmach) [inline src/pages/Search.jsx:1]: remove comments

File: src/pages/Search.jsx [hunk 5]
```diff
@@ -1,10 +1,46 @@
...
+import styled from "styled-components"
+
 import Layout from "../containers/Layout"
 import SearchContainer from "../containers/Search"
+// import SearchResult from "./SearchResult"
```
Comment (alexandrtovmach) [inline src/pages/Search.jsx:6]: remove comments

File: src/pages/SearchResult.jsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+              <Arrow>&#8592;</Arrow>
+            </BackBtn>
+          </Link>
+          {!incomingData.length && <Spinner>Loading . . .</Spinner>}
+          {incomingData
+            .filter(
+              ({ departureCity, arriveCity, departureDate }) =>
+                departureCity.includes(departureCityInput) &&
+                arriveCity.includes(arrivalCityInput) &&
+                new Date(departureDate) >= new Date(departureStartDateInput) &&
```
Comment (alexandrtovmach) [inline src/pages/SearchResult.jsx:84]:
```suggestion
                (!departureStartDateInput || new Date(departureDate) >= new Date(departureStartDateInput)) &&
```

File: src/pages/SearchResult.jsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                new Date(departureDate) <= new Date(departureEndDateInput)
```
Comment (alexandrtovmach) [inline src/pages/SearchResult.jsx:85]:
```suggestion
                (!departureEndDateInput || new Date(departureDate) <= new Date(departureEndDateInput))
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
- _src/MOCK.jsx_ - should be deleted
- if no date range provided, show all results based on cities only

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**